internal/pkg/dependencies/abstract.go

```diff
@@ -0,0 +1,54 @@
+package dependencies
+
+import (
+	"context"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/encryption"
+	"github.com/keboola/keboola-as-code/internal/pkg/event"
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/project"
+	"github.com/keboola/keboola-as-code/internal/pkg/remote"
+	"github.com/keboola/keboola-as-code/internal/pkg/scheduler"
+	"github.com/keboola/keboola-as-code/internal/pkg/template"
+	"github.com/keboola/keboola-as-code/internal/pkg/template/repository"
+	loadState "github.com/keboola/keboola-as-code/pkg/lib/operation/state/load"
+)
+
+// Container contains dependencies for all use-cases.
+type Container interface {
+	AbstractDeps
+	CommonDeps
```

Comment:
Tento interface je zlozeny z 2:
- `AbstractDeps` su specificke pre CLI, napr. nacitanie projektu, template ...., v testoch su mockovane
- `CommonDeps` su spolocne pre CLI, testy (aj webservice) ... napr. v `AbstractDeps` je `StorageApiHost` a v `CommonDeps` je `StorageApi`, tj. ako pomocou hostu, vytvorit StorageAPI

Ono by to mohol byt aj jeden interface, je to tak kvoli nazornosti.